util/intest/assert.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build intest
+
+package intest
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// Assert asserts a condition. It only works in test (intest.InTest == true).
+// You can assert a condition like this to assert a variable `foo` is not nil: `assert.Assert(foo != nil)`.
+// Or you can pass foo as a parameter directly for simple: `assert.Assert(foo)`
+// You can also assert a function that returns a bool: `intest.Assert(func() bool { return foo != nil })`
+// If you pass a function without a signature `func() bool`, the function will always panic.
+func Assert(cond any, msgAndArgs ...any) {
+	if InTest {
```

Comment:
we can rely on `InTest` is always true when `//go:build intest`, or to make it more obvious, merge assert.go and intest.go into one file.

Reply:
I prefer to leave it here. Some PR are trying to use `testing.Testing()` to replace `inTest` flag. Though this PR cannot merge because of some bazel bug now, but one day it should merge to master, it's a good way to make sure this place can be modified in right way.

Reply:
We can still eliminated the Assert function in build phase at that time if we want. just to add a new flag `inRelease` when building a product version...

Reply:
> Some PR are trying to use testing.Testing() to replace inTest flag

Do you expect Assert should be enabled for testing.Testing()? With the "intest" build tag Assert is no-op.

> just to add a new flag inRelease when building a product version...

makes me think intest and indebug/inrelease are two separate things.

Reply:
```suggestion
Do you expect Assert should be enabled for testing.Testing()? With the "intest" build tag Assert is no-op.
```

Yes，at that time intest flag may be removed.

```
makes me think intest and indebug/inrelease are two separate things.
```

At that time, `inrelease` tag will be used to optimize some performance like `-O` in C, but it may not be very necessary because I don't think it has a very big performance affect.